### PR TITLE
CNI-11067; Add fastlane for releasing;

### PR DESCRIPTION
**Pull request by jotshin**
JIRA ticket: https://conichi.atlassian.net/browse/CNI-11067

## Description
- Add fastlane files for auto release.
- Run the script until bumping the podspecs version to check the functionality.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+platform :ios do
+
+  before_all do
+    skip_docs
+    import_shared_fastfile
+  end
+
+  ######################### PUBLIC LANES ##########################
+
+  desc "Release a new version of the ios-sdk. Uploading a new release to a GitHub and CocoaPods trunk"
```
Comment (David-Henner) [inline fastlane/Fastfile]: `Release a new version of the ios-sdk` text here is probably wrong

File: fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    make_github_release(version: upcoming_version)
+
+    podspecs_push
+  end
+
+  ######################### PRIVATE LANES ##########################
+
+  desc "Build frameworks with Carthage"
+  private_lane :build_frameworks do
+    framework_name = "#{ENV['ITININERARIES_FRAMEWORK']}.framework.zip"
```
Comment (David-Henner) [inline fastlane/Fastfile]: I think there's a typo `ITININERARIES_FRAMEWORK`
Should be `ITINERARIES_FRAMEWORK`

**Review by David-Henner — COMMENTED** [review]
Thanks for the great work 👍

File: fastlane/Fastfile (hunk 1 above)
Comment (jotshin) [reply fastlane/Fastfile]: Thanks and updated in 1e4ec1c.

File: fastlane/Fastfile (hunk 2 above)
Comment (jotshin) [reply fastlane/Fastfile]: Nice catch. Thanks and updated in 1e4ec1c.

File: fastlane/Fastfile [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      output: framework_name,
+      command: "archive"
+    )
+
+    [framework_name]
+  end
+
+  desc "Push podspecs to conichi specs repo"
+  private_lane :podspec_push do
+  	pod_push(path: ENV['ITININERARIES_PODSPEC'], repo: ENV['CONICHI_PODS_REPO_ORGANIZATION'], sources: [ENV['PRIVATE_CONICHI_PODS_SPECS'], ENV['COCOAPODS_MASTER_SPECS']], allow_warnings: true)
```
Comment (David-Henner) [inline fastlane/Fastfile]: `ITININERARIES_PODSPEC` typo still present

File: fastlane/Fastfile [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      tag_name: tag_name,
+      api_token: ENV['CONICHI_CI_GITHUB_TOKEN'],
+      upload_assets: assets
+    )
+  end
+
+  desc "Bump podspec version"
+  private_lane :podspecs_version_bump do |options|
+    version = options[:version]
+    version_bump_podspec(path: ENV['ITININERARIES_PODSPEC'], version_number: version)
```
Comment (David-Henner) [inline fastlane/Fastfile]: Same here

File: fastlane/Fastfile (hunk 3 above)
Comment (jotshin) [reply fastlane/Fastfile]: Sorry for missing this. Updated in 1221674.

File: fastlane/Fastfile (hunk 4 above)
Comment (jotshin) [reply fastlane/Fastfile]: Thanks and updated in 1221674.

**Review by David-Henner — APPROVED** [review]

**Pull request merged**